src/deluge/util/d_string.h

```diff
@@ -72,3 +73,31 @@ class String {
 
 	char* stringMemory;
 };
+
+/// A stack based string buffer with utility functions to append and format contents.
+template <std::size_t N>
```

Comment:
This class also adds a noticeable amount of bin size - rather than templating on the size you could use a private size member set at construction, that will avoid duplicating the functions for each new length of string